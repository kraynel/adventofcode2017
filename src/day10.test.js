const { computeHash, computeFullHash } = require("./day10");
const fs = require("fs");

describe("computeHash", () => {
  it("should compute correct hash", () => {
    expect(computeHash(5, "3")).toBe(2);
    expect(computeHash(5, "3, 4")).toBe(12);
    expect(computeHash(5, "3, 4, 1")).toBe(12);
    expect(computeHash(5, "3, 4, 1, 5")).toBe(12);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day10.txt", "utf8", (err, data) => {
      expect(computeHash(256, data)).toMatchSnapshot();
      done();
    });
  });
});

describe("computeFullHash", () => {
  it("should compute full hash", () => {
    expect(computeFullHash("")).toBe("a2582a3a0e66e6e86e3812dcb672a272");
    expect(computeFullHash("AoC 2017")).toBe(
      "33efeb34ea91902bb2f59c9920caa6cd"
    );
    expect(computeFullHash("1,2,3")).toBe("3efbe78a8d82f29979031a4aa0b16a9d");
    expect(computeFullHash("1,2,4")).toBe("63960835bcdc130f0b66d7ff4f6a5a8e");
  });

  it("should get the value", done => {
    fs.readFile("inputs/day10.txt", "utf8", (err, data) => {
      expect(computeFullHash(data)).toMatchSnapshot();
      done();
    });
  });
});
