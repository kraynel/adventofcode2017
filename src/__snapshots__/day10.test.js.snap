// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`computeFullHash should get the value 1`] = `"44f4befb0f303c0bafd085f97741d51d"`;

exports[`computeHash should get the value 1`] = `5577`;
